backend/src/graphql.ts

```diff
@@ -27,10 +31,49 @@ schema.objectType({
     t.model.title()
     t.model.body()
     t.model.author()
-    t.model.published()
+    t.model.status()
+    t.model.language({ type: 'Language' })
   },
 })
 
+schema.objectType({
+  name: 'Location',
+  definition(t) {
+    t.model.id()
+    t.model.country()
+    t.model.city()
+  },
+})
+
+schema.objectType({
+  name: 'Language',
+  definition(t) {
+    t.model.id()
+    t.model.name()
+    t.model.posts()
+    t.model.dialect()
+    t.model.nativeUsers()
+    //t.model.learningUsers()
+    t.field('learningUsers', {
+      list: true,
+      type: 'User',
+      resolve: async (language, _args, ctx) => {
+        const result = await ctx.db.language
+          .findOne({ where: { id: language.id } })
+          .learningUsers({ include: { user: true } })
+        return result.map((r) => r.user)
+      },
+    })
+  },
+})
+
+// schema.objectType({
+//   name: 'LanguageLearned',
+//   definition(t) {
+//     t.
+//   }
+// })
```

Comment:
@nlitwin just as a reminder I'm going to be cleaning all of this up and fixing it up piece by piece in a series of coming PRs that will each be specific and contained 😊